src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -358,60 +358,47 @@ APPLICATION_INFO::FindRequestHandlerAssembly(STRU& location)
 
 HRESULT
 APPLICATION_INFO::FindNativeAssemblyFromGlobalLocation(
-    PCWSTR libraryName,
-    STRU* struFilename)
+    PCWSTR pstrHandlerDllName,
+    STRU* struFilename
+)
 {
     HRESULT hr = S_OK;
-    DWORD dwSize = MAX_PATH;
-    BOOL  fDone = FALSE;
-    DWORD dwPosition = 0;
-
-    // Though we could call LoadLibrary(L"aspnetcorerh.dll") relying the OS to solve
-    // the path (the targeted dll is the same folder of w3wp.exe/iisexpress)
-    // let's still load with full path to avoid security issue
-    if (FAILED(hr = struFilename->Resize(dwSize + 20)))
-    {
-        goto Finished;
-    }
 
-    while (!fDone)
+    try
     {
-        DWORD dwReturnedSize = GetModuleFileNameW(g_hModule, struFilename->QueryStr(), dwSize);
-        if (dwReturnedSize == 0)
-        {
-            hr = HRESULT_FROM_WIN32(GetLastError());
-            fDone = TRUE;
-            goto Finished;
-        }
-        else if ((dwReturnedSize == dwSize) && (GetLastError() == ERROR_INSUFFICIENT_BUFFER))
-        {
-            dwSize *= 2; // smaller buffer. increase the buffer and retry
-            if (FAILED(hr = struFilename->Resize(dwSize + 40))) // + 40 for aspnetcorerh.dll
-            {
-                goto Finished;
-            }
-        }
-        else
+        std::wstring modulePath = GlobalVersionUtility::GetModuleName(g_hModule);
+
+        modulePath = GlobalVersionUtility::RemoveFileNameFromFolderPath(modulePath);
+
+        std::wstring retval = GlobalVersionUtility::GetGlobalRequestHandlerPath(modulePath.c_str(),
+            m_pConfiguration->QueryHandlerVersion()->QueryStr(),
+            pstrHandlerDllName
+        );
+
+        if (FAILED(hr = struFilename->Copy(retval.c_str())))
         {
-            fDone = TRUE;
+            return hr;
         }
     }
-
-    if (FAILED(hr = struFilename->SyncWithBuffer()))
+    catch (std::exception& e)
     {
-        goto Finished;
+        STRU struEvent;
+        if (SUCCEEDED(struEvent.CopyA(e.what())))
```

Comment:
you may want to give more info in this error log, e.g., application information and path tried to searched. It will help user to identify the root cause.